algo/src/test/java/com/hartwig/serve/sources/ckb/ActionableTrialFactoryTest.java

```diff
@@ -0,0 +1,128 @@
+package com.hartwig.serve.sources.ckb;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.Lists;
+import com.hartwig.serve.ckb.datamodel.CkbEntry;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.ClinicalTrial;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.ImmutableLocation;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.ImmutableVariantRequirementDetail;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.Location;
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.VariantRequirementDetail;
+import com.hartwig.serve.datamodel.EvidenceDirection;
+import com.hartwig.serve.datamodel.EvidenceLevel;
+import com.hartwig.serve.datamodel.Knowledgebase;
+
+import org.apache.commons.compress.utils.Sets;
+import org.jetbrains.annotations.NotNull;
+import org.junit.Test;
+
+public class ActionableTrialFactoryTest {
+
+    @Test
+    public void canCreateActionableTrial() {
+        Location location = ImmutableLocation.builder().nctId("nctid").city("").country("Netherlands").status("Recruiting").build();
+        VariantRequirementDetail requirementType =
+                ImmutableVariantRequirementDetail.builder().profileId(0).requirementType("required").build();
+        CkbEntry entry = CkbTestFactory.createEntryWithClinicalTrial(location, "Recruiting", requirementType);
+        ActionableTrialFactory actionableTrialFactory = new ActionableTrialFactory();
+        Set<ActionableEntry> entrySet = actionableTrialFactory.create(entry, "KRAS", "gene");
+
+        assertEquals(1, entrySet.size());
```

Comment:
I haven't tested this but suspect your logic only works because both the variant requirement detail and entry have profileId 0. This is explicitly set in requirementType but not in entry, so this is an unwanted hidden dependency.

Reply:
I changed this to set it explicitly in both